driver-core/src/test/unit/com/mongodb/DBObjectCodecSpecification.groovy

```diff
@@ -59,8 +59,9 @@ import static org.bson.codecs.configuration.CodecRegistries.fromRegistries
 class DBObjectCodecSpecification extends Specification {
 
     def bsonDoc = new BsonDocument()
-    def codecRegistry = fromProviders([new ValueCodecProvider(), new DBObjectCodecProvider(), new BsonValueCodecProvider()])
-    def dbObjectCodec = new DBObjectCodec(codecRegistry)
+    def codecRegistry = fromRegistries(fromCodecs(new UuidCodec(JAVA_LEGACY)),
+            fromProviders([new ValueCodecProvider(), new DBObjectCodecProvider(), new BsonValueCodecProvider()]))
+    def dbObjectCodec = new DBObjectCodec(codecRegistry).withUuidRepresentation(JAVA_LEGACY)
```

Comment:
Do you need the withUuidRepresentation extra call? Or is it needed because the registry wasn't used to get the codec?eg: `codecRegistry.get(DBObjectCodec.class)`

Reply:
Yes, exactly.  And there is currently no way to configure the DbObjectCodecProvider with a uuid representation.

Reply:
Ack